BJS2-69739

KolyanGrom opened the pull request:
Решение задачи "Разбираем склад Amazon"

Commit:
BJS2-69739

Решение задачи "Разбираем склад Amazon"

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/Main.java:
```diff
@@ -0,0 +1,30 @@
+
+import java.util.List;
+
+@Slf4j
```
Оооо, использовал @Slf4j, круто

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/Product.java:
```diff
@@ -0,0 +1,23 @@
+    private final String id;
+
+    public Product(String name, Category category) {
+        this.name = Objects.requireNonNull(name, "Product name cannot be null");
```
Хорошо, что проверки делал. И идея с uuid то же хорошая

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/ProductManager.java:
```diff
@@ -0,0 +1,73 @@
+
+    public boolean removeProduct(Category category, String name) {
+        Objects.requireNonNull(category, "Category cannot be null");
+        if (name == null || name.trim().isEmpty()) {
```
Это условие повторяется несколько раз, так же как и проверка категории. Имеет смысл вынести их в отдельный метод, и использовать этот метод везде, где нужна валидация

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/ProductManager.java:
```diff
@@ -0,0 +1,73 @@
+            throw new IllegalArgumentException("Product name cannot be null or empty");
+        }
+
+        Product newProduct = new Product(name.trim(), category);
```
Несколько раз делаешь операцию .trim().
Можно в начале метода создать переменную например `String trimmedName = name.trim()`. И уже trimmedName использовать дальше.
Так же в след. методе

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/ProductManager.java:
```diff
@@ -0,0 +1,73 @@
+    public List<Product> findProductsByCategory(Category category) {
+        Objects.requireNonNull(category, "Category cannot be null");
+
+        return products.stream()
```
Здорово, что используешь стримы. Старайся переменные и простыми описания ми не создавать, например p. 
Лучше называть понятными именами. На этом примере сложно запутаться, но когда кода больше и внутри стримов многое происходит, это реально больно, разобраться что к чему, когда одни p, f, k, z переменные.
`.filter(product -> product.getCategory() == category)`

Commit:
BJS2-68987

Исправление замечаний.
К сожалению не понял как это реализовать: "Это условие (if (trimmedName.isEmpty()) {
                                                                    throw new IllegalArgumentException("Product name cannot be null or empty");) повторяется несколько раз, так же как и проверка категории. Имеет смысл вынести их в отдельный метод, и использовать этот метод везде, где нужна валидация"

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/ProductManager.java:
```diff
@@ -0,0 +1,75 @@
+        }
+        return products.removeIf(product ->
+                product.getCategory() == category
+                        && product.getName().equalsIgnoreCase(name.trim())
```
name.trim() -> trimmedName

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/ProductManager.java:
```diff
@@ -0,0 +1,75 @@
+    public Product addProduct(Category category, String name) {
+        String trimmedName = name.trim();
+        Objects.requireNonNull(category, "Category cannot be null");
+        if (name.trim().isEmpty()) {
```
name.trim() -> trimmedName

SnurnicynKonstantin commented on src/main/java/school/faang/parse_the_warehouse_amazon/ProductManager.java:
```diff
@@ -0,0 +1,75 @@
+        String trimmedName = name.trim();
+        Objects.requireNonNull(category, "Category cannot be null");
+        if (trimmedName.isEmpty()) {
+            throw new IllegalArgumentException("Product name cannot be null or empty");
```
Комментарий к вынесению валидации в отдельный метод.
Можно создать метод:
```
private viod validateName(String name) {
    if (name) {
            throw new IllegalArgumentException("Product name cannot be null or empty");
    }
}
```

И использовать его здесь вместо условия и в методе выше:
`validateName(trimmedName);`

Commit:
BJS2-69739

Исправление замечаний

Commit:
BJS2-69739

Исправление замечаний проверки checkstyleMain

SnurnicynKonstantin reviewed (approved):
Что используешь стримы это здорово
